src/main/xqy/qc-workspace-importer.xqy

```diff
@@ -0,0 +1,100 @@
+        declare variable $xquery-query-template as xs:string external;
+        declare variable $workspace as element(export) external;
+        declare variable $user as xs:string external;
+        let $_ := xdmp:log(text{("user", $user)})
+        let $wsid := xdmp:random()
+        let $imported-wsname := string($workspace/workspace/@name)
+        let $existing-wsnames := amped-qconsole:qconsole-get-user-workspaces(())/qconsole:name/string()
+        let $wsname :=
+            if( $imported-wsname = $existing-wsnames )
+            then qconsole-model:generate-workspace-name(())
+            else $imported-wsname
+        let $queries := $workspace/workspace/query
+        let $userid := xdmp:user($user)
+        let $_ := xdmp:log(text{("userid", $userid)})
+        let $ws :=  <qconsole:workspace>
+                        <qconsole:id>{$wsid}</qconsole:id>
+                        <qconsole:name>{$wsname}</qconsole:name>
+                        <qconsole:security>
+                            <qconsole:userid>{$userid}</qconsole:userid>
+                        </qconsole:security>
+                        <qconsole:active>true</qconsole:active>
+                        <qconsole:queries>
+                            {
+                            for $q at $i in $queries
+                            let $qid := xdmp:random()
+                            let $qname := string($q/@name)
+                            let $focus := string($q/@focus)
+                            let $active := string($q/@active)
+                            let $content-source :=
+                                if ( exists($q/@content-source) )
+                                then string($q/@content-source)
+                                else qconsole-model:default-content-source()
+                            let $mode := string($q/@mode)
+                            let $query-text := text { $q }
+                            let $q-uri := concat("/queries/", $qid, ".txt")
+                            let $save-q := amped-qconsole:qconsole-document-insert($q-uri, $query-text)
+                            return
+                            <qconsole:query>
+                                <qconsole:id>{$qid}</qconsole:id>
+                                <qconsole:name>{$qname}</qconsole:name>
+                                <qconsole:content-source>{$content-source}</qconsole:content-source>
+                                <qconsole:active>{$active}</qconsole:active>
+                                <qconsole:focus>{$focus}</qconsole:focus>
+                                <qconsole:mode>{$mode}</qconsole:mode>
+                            </qconsole:query>
+                            }
+                        </qconsole:queries>
+                    </qconsole:workspace>
+        let $ws-uri := concat("/workspaces/", $wsid, ".xml")
+        let $save-ws := amped-qconsole:qconsole-document-insert($ws-uri, $ws)
```

Comment:
Similar question as above - can a workspace file be imported as-is? I don't know what happens if the IDs don't match real user/database resources. I'll try that out locally, this is new to me.